fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/impl/FHIRPersistenceJDBCImpl.java

```diff
@@ -1452,32 +1458,114 @@ public MultiResourceResult history(FHIRPersistenceContext context, Class<? exten
 
         List<ResourceResult<? extends Resource>> resourceResults = new ArrayList<>(resourceDTOList.size());
         try {
-            for (com.ibm.fhir.persistence.jdbc.dto.Resource resourceDTO : resourceDTOList) {
-                // TODO Linear fetch of a large number of resources will extend response times. Need
-                // to look into batch or parallel fetch requests
-                ResourceResult<? extends Resource> resourceResult = convertResourceDTOToResourceResult(resourceDTO, resourceType, elements, includeResourceData);
-                
-                // Check to make sure we got a Resource if we asked for it and expect there to be one
+            if (this.payloadPersistence != null) {
+                // With offloading, we can use async reads to fetch the payloads in parallel, making this
+                // significantly quicker for large bundles (search, history etc)
+                fetchPayloadsForDTOList(resourceResults, resourceDTOList, resourceType, elements, includeResourceData);
+            } else {
+                for (com.ibm.fhir.persistence.jdbc.dto.Resource resourceDTO : resourceDTOList) {
+                    // TODO Linear fetch of a large number of resources will extend response times. Need
```

Comment:
Is this TODO intended to be addressed now or in the future?

Reply:
comment no longer applied. removed